iphone/Maps/Bookmarks/Categories/BMCViewModel/BMCDefaultViewModel.swift

```diff
@@ -121,8 +124,8 @@ extension BMCDefaultViewModel {
 
     let category = categories[index]
     categories.remove(at: index)
-    manager.deleteCategory(category.categoryId)
     view?.delete(at: [IndexPath(row: index, section: section)])
+    manager.deleteCategory(category.categoryId)
```

Comment:
Why the order here is important?

Reply:
When the `manager.deleteCategory(category.categoryId)` is called the bookmarks manager will trigger the observers that they should update their content. And the 
```
func onBookmarksCategoryDeleted(_ groupId: MWMMarkGroupID) {
    reloadData()
  }
```
will trigger the reloading of the table (and remove the row).
BUT 
`view?.delete(at: [IndexPath(row: index, section: section)])` also removes he row.

And if `manager.deleteCategory(category.categoryId)` is called first (and reloads the data), the `view?.delete(at: [IndexPath(row: index, section: section)])` will attempt to delete the element that _is already deleted_ ant it crashes the app.

So it is important to  reorder this lines to update the local data source in `categories.remove(at: index)`, and only after that update the core.

Reply:
Question: is `view?.delete(at: [IndexPath(row: index, section: section)])` call needed in this case?

Reply:
Its needed for the _beautiful deletion animation_ instead of just reloading with fade.

Reply:
A simpler, faster version without any side effects may be better than some animation, no? )

Reply:
Looks a little bit rough...

![Simulator Screen Recording - iPhone 15 Pro - 2024-05-22 at 11 10 21](https://github.com/organicmaps/organicmaps/assets/79797627/10444d48-7de7-459e-bba1-d7b98ab0fe31)

Reply:
No big difference, ok, let's use it as is.